mpi4jax/collective_ops/sendrecv.py

```diff
@@ -18,16 +18,27 @@
     wrap_as_hashable,
     unpack_hashable,
     default_primitive_impl,
+    HashableMPIType,
 )
 
 from ..warn import warn_missing_omnistaging
+from ..validation import enforce_types
 
 # The Jax primitive
 mpi_sendrecv_p = Primitive("sendrecv_mpi")  # Create the primitive
 mpi_sendrecv_impl = default_primitive_impl(mpi_sendrecv_p)
 
 
 # This function applies the primitive to an AST
+@enforce_types(
+    source=int,
+    dest=int,
+    sendtag=int,
+    recvtag=int,
+    comm=(_MPI.Intracomm, HashableMPIType),
+    status=(type(None), _MPI.Status),
```

Comment:
`status` is never passed as an hashable type?

Reply:
Good catch! This has to be hashable, too.